libsentrykube/utils.py

```diff
@@ -290,57 +291,64 @@ def ensure_libsentrykube_folder() -> Path:
 
 
 @cache
-def ensure_kubectl(version: str = KUBECTL_VERSION) -> Path:
+def ensure_kubectl(
+    binary: str = KUBECTL_BINARY, version: str = KUBECTL_VERSION
+) -> Path:
     base = ensure_libsentrykube_folder() / "kubectl" / f"v{version}"
-    path = base / "kubectl"
+    path = base / binary
     if path.is_file():
         return path
 
-    base.mkdir(parents=True, exist_ok=True)
+    if binary == "kubectl":
```

Comment:
nit: invert the iff and raise the error `if binary != "kubctl". The result is a large block that is less nested and more readable. The else block only raises the exception

Reply:
My plan was to handle `kubecolor` installs in another PR down the road which means this will turn into an `if` ... `elif` ... `else` situation.

Reply:
Good idea, I had plans for auto-install in the future which I was thinking of turning this into an `if` ... `elif` ... situation but `kubecolor` still needs `kubectl` installed anyways so we basically need to run that block in all circumstances.